frontend/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -557,10 +406,48 @@ function DataFrame({
               setIsFocused(true)
             }
           }}
+          // Add shadow for index columns and header on scroll:
+          fixedShadowX={true}
+          fixedShadowY={true}
+          // onPaste is deactivated in the read-only mode:
+          onPaste={false}
           experimental={{
             // We use an overlay scrollbar, so no need to have space for reserved for the scrollbar:
             scrollbarWidthOverride: 1,
           }}
+          // Add support for additional cells:
+          customRenderers={extraCellArgs.customRenderers}
+          // If element is editable, add additional properties:
+          {...(!showEmptyState &&
+            element.editingMode !== ArrowProto.EditingMode.READ_ONLY &&
+            !disabled && {
+              // Support fill handle for bulk editing
+              fillHandle: true,
+              // Support editing:
+              onCellEdited,
+              // Support pasting data for bulk editing:
+              onPaste,
+              // Support deleting cells & rows
```

Comment:
super nit: can add a ":" at the end of commenting to make it consistent.

Reply:
changed it 👍